server/db/common/common.go

```diff
@@ -23,41 +23,39 @@ func SelectEarliestUpdatedSubs(subs []t.Subscription, opts *t.QueryOpt, maxResul
 		}
 	}
 
-	if !ims.IsZero() || len(subs) > limit {
-		// Now that we fetched potentially more subscriptions than needed, we got to take those with the oldest modifications.
-		// Sorting in ascending order by modification time.
-		sort.Slice(subs, func(i, j int) bool {
-			return subs[i].UpdatedAt.Before(subs[j].UpdatedAt)
+	// No cache management and the number of results is below the limit: return all.
+	if ims.IsZero() && len(subs) <= limit {
+		return subs
+	}
+
+	// Now that we fetched potentially more subscriptions than needed, we got to take those with the oldest modifications.
+	// Sorting in ascending order by modification time.
+	sort.Slice(subs, func(i, j int) bool {
+		return SelectLatestTime(subs[i].UpdatedAt, subs[i].GetTouchedAt()).
```

Comment:
Perhaps for better readbility, make a helper function for
`SelectLatestTime(s.UpdatedAt, s.GetTouchedAt())`, e.g. `LastModified`.